### PR TITLE
feat: support code readability evals

### LINKED ISSUE #1327: [evals] prompt template for Code Gen and notebook
Team Code Gen:
Code Eval: Goal is to evaluate the correctness and quality of generated code. We suggest this team come up with some approaches to evaluating code, correctness, quality, risk, readability etc… The Eval should give you a good or bad response. You can have more than 1 Eval but it should only give you a binary output.
 
Datasets: The dataset human eval from huggingface can be used to derive a dataset for testing. It contains descriptions of coding problems and answers. We want a mixture of correct and incorrect (problematic) responses. What we want is to insert mistakes or issues into the coding datasets 50% of the time. You probably need to have GPT help generate the issues.    
 
https://huggingface.co/datasets/openai_humaneval

### PR DESCRIPTION
**Pull request by harrisonchu**
resolves #1327

### CODE REVIEW

**Comment by harrisonchu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by harrisonchu** [comment]
recheck"

File: src/phoenix/experimental/evals/templates/default_templates.py [hunk 1]
```diff
@@ -38,3 +38,23 @@
     # Answer: {response}
     Is the answer above factual or hallucinated based on the query and reference text?
 """
+
+CODE_READABILITY_PROMPT_TEMPLATE_STR = """
+You are a stern but practical senior software engineer who cares a lot about simplicity and
```
Comment (axiomofjoy) [inline src/phoenix/experimental/evals/templates/default_templates.py:43]: Stern but practical 😂

File: src/phoenix/experimental/evals/templates/default_templates.py [hunk 2]
```diff
@@ -38,3 +38,23 @@
     # Answer: {response}
     Is the answer above factual or hallucinated based on the query and reference text?
 """
+
+CODE_READABILITY_PROMPT_TEMPLATE_STR = """
```
Comment (axiomofjoy) [inline src/phoenix/experimental/evals/templates/default_templates.py:42]: Can you also add a map called `CODE_READABILITY_PROMPT_OUTPUT_MAP`? It should be used as a mapping to transform the values in the "readability" column from booleans to string values.

See Nate's PR for reference: https://github.com/Arize-ai/phoenix/pull/1338/files

Comment (axiomofjoy) [reply src/phoenix/experimental/evals/templates/default_templates.py:42]: It should look something like:

```python
CODE_READABILITY_PROMPT_OUTPUT_MAP = {
    True: "readable", False: "unreadable"
}
```

Comment (axiomofjoy) [reply src/phoenix/experimental/evals/templates/default_templates.py:42]: See this comment for context: https://github.com/Arize-ai/phoenix/pull/1338#issuecomment-1720383668

**Pull request merged**